security-council-rehearsals/templates/r5-guardian-override/README.md

```diff
@@ -0,0 +1,297 @@
+with your wallet which will be collected by a Facilitator, who will
+submit all signatures and perform the execution onchain.
+
+
+## Approving the transaction
+
+### 1. Update repo and move to the appropriate folder for this rehearsal task:
+
+```
+cd superchain-ops
+git pull
+just install
+cd security-council-rehearsals/$(REPLACE_WITH_REHEARSAL_FOLDER)
+```
+
+### 2. Setup Ledger
+
+Your Ledger needs to be connected and unlocked. The Ethereum
+application needs to be opened on Ledger with the message "Application
+is ready".
+
+### 3. Simulate and validate the transaction
+
+Make sure your ledger is still unlocked and run the following.
+
+Remember that by default just is running with the address derived from
+`/0` (first nonce). If you wish to use a different account, append the value `X` to the end of the following command where X is the derivation path of the address that you want to use.
+
+``` shell
+just \
+   --dotenv-path $(pwd)/.env \
+   --justfile ../../single.just \
+simulate # 0 or 1 or your derivation path index
+```
+
+You will see a "Simulation link" from the output.
+
+Paste this URL in your browser. A prompt may ask you to choose a
+project, any project will do. You can create one if necessary.
+
+Click "Simulate Transaction".
+
+We will be performing 3 validations and extract the domain hash and
+message hash to approve on your Ledger:
+
+1. Validate integrity of the simulation.
+2. Validate correctness of the state diff.
+3. Validate and extract domain hash and message hash to approve.
+
+#### 3.1. Validate integrity of the simulation.
+
+Make sure you are on the "Overview" tab of the tenderly simulation, to
+validate integrity of the simulation, we need to
+
+1. "Network": Check the network is Ethereum Mainnet.
+2. "Timestamp": Check the simulation is performed on a block with a
+   recent timestamp (i.e. close to when you run the script).
+3. "Sender": Check the address shown is your signer account.
+
+![](./images/tenderly-overview-network.png)
+
+#### 3.2. Validate correctness of the state diff.
+
+Now click on the "State" tab. Verify that:
+
+**In the State Overrides:**
+
+There is a single override to the rehearsal safe address (`$COUNCIL_SAFE`), which
+sets
+
+- **Key:** `0x0000000000000000000000000000000000000000000000000000000000000004` <br/>
+  **Value:** `0x0000000000000000000000000000000000000000000000000000000000000001` <br/>
+   **Meaning:** Enables the simulation by setting the threshold to 1. The key can be validated by the location of the `threshold` variable in the [Safe's Storage Layout](https://github.com/safe-global/safe-smart-account/blob/v1.3.0/contracts/examples/libraries/GnosisSafeStorage.sol#L14).
+
+**In the State Changes:**
+
+The Guardian Safe (`$GuardianSafe_ADDRESS`), will having the following changes:
+
+1. the `nonce` will be increased from `2` to `4`. This is increased by two because the
+   Safe has made two separate calls.
```

Comment:
Interesting, why isn't it just a single multicall with a nonce increment of 1?

Reply:
This ceremony is done as a single ceremony, meaning there is a multicall with two tx from the SC Safe. Since both of those tx go to the Guardian Safe, it has two nonce changes.

If we want to do the multicall from the Guardian Safe, we could use the `nested.just` script instead, but then we would have to do the multi-step `approve` and `execute` and I imagine that other complications would arise which I have not yet considered.